server/coordinator/procedure/drop_partition_table.go

```diff
@@ -0,0 +1,421 @@
+	stateClosePartitionTables     = "StateClosePartitionTables"
+	stateDropPartitionTableFinish = "StateDropPartitionTableFinish"
+)
+
+var (
+	createDropPartitionTableEvents = fsm.Events{
+		{Name: eventDropDataTable, Src: []string{stateDropPartitionTableBegin}, Dst: stateDropDataTable},
+		{Name: eventDropPartitionTable, Src: []string{stateDropDataTable}, Dst: stateDropPartitionTable},
+		{Name: eventClosePartitionTables, Src: []string{stateDropPartitionTable}, Dst: stateClosePartitionTables},
+		{Name: eventDropPartitionTableFinish, Src: []string{stateClosePartitionTables}, Dst: stateDropPartitionTableFinish},
+	}
+	createDropPartitionTableCallbacks = fsm.Callbacks{
+		eventDropDataTable:            dropDataTablesCallback,
+		eventDropPartitionTable:       dropPartitionTableCallback,
+		eventClosePartitionTables:     closePartitionTableCallback,
+		eventDropPartitionTableFinish: finishDropPartitionTableCallback,
+	}
+)
+
+type DropPartitionTableProcedure struct {
+	id       uint64
+	fsm      *fsm.FSM
+	cluster  *cluster.Cluster
+	dispatch eventdispatch.Dispatch
+	storage  Storage
+
+	onSucceeded func(cluster.TableInfo) error
+	onFailed    func(error) error
+
+	req *metaservicepb.DropTableRequest
+
+	lock  sync.RWMutex
+	state State
+}
+
+type DropPartitionTableProcedureRequest struct {
+	ID          uint64
+	Cluster     *cluster.Cluster
+	Dispatch    eventdispatch.Dispatch
+	Storage     Storage
+	Req         *metaservicepb.DropTableRequest
+	OnSucceeded func(result cluster.TableInfo) error
+	OnFailed    func(error) error
+}
+
+func NewDropPartitionTableProcedure(request DropPartitionTableProcedureRequest) *DropPartitionTableProcedure {
+	fsm := fsm.NewFSM(
+		stateDropPartitionTableBegin,
+		createDropPartitionTableEvents,
+		createDropPartitionTableCallbacks,
+	)
+	return &DropPartitionTableProcedure{
+		id:          request.ID,
+		fsm:         fsm,
+		cluster:     request.Cluster,
+		dispatch:    request.Dispatch,
+		storage:     request.Storage,
+		req:         request.Req,
+		onSucceeded: request.OnSucceeded,
+		onFailed:    request.OnFailed,
+	}
+}
+
+func (p *DropPartitionTableProcedure) ID() uint64 {
+	return p.id
+}
+
+func (p *DropPartitionTableProcedure) Typ() Typ {
+	return DropPartitionTable
+}
+
+func (p *DropPartitionTableProcedure) Start(ctx context.Context) error {
+	p.updateStateWithLock(StateRunning)
+
+	dropPartitionTableRequest := &dropPartitionTableCallbackRequest{
+		ctx:         ctx,
+		cluster:     p.cluster,
+		dispatch:    p.dispatch,
+		sourceReq:   p.req,
+		onSucceeded: p.onSucceeded,
+		onFailed:    p.onFailed,
+	}
+
+	for {
+		switch p.fsm.Current() {
+		case stateDropPartitionTableBegin:
+			if err := p.persist(ctx); err != nil {
+				return errors.WithMessage(err, "drop partition table procedure persist")
+			}
+			if err := p.fsm.Event(eventDropDataTable, dropPartitionTableRequest); err != nil {
+				p.updateStateWithLock(StateFailed)
+				return errors.WithMessagef(err, "drop partition table procedure")
```

Comment:
```suggestion
				return errors.WithMessage(err, "drop partition table procedure")
```
And check other places in this file.